python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,387 @@
+    False, or if MySQL error occurs.
+    """
+    dbNameStr = "`%s`." % dbName if dbName is not None else ""
+    try:
+        conn.execute("CREATE TABLE %s`%s` %s" % \
+                     (dbNameStr, tableName, tableSchema))
+    except DBAPIError as e:
+        if e.orig[0] == MySqlErr.ER_DB_DROP_EXISTS:
+            raise DbException(DbException.TB_EXISTS, dbNameStr)
+        elif e.orig[0] == MySqlErr.ER_NO_DB_ERROR:
+            raise DbException(DbException.INVALID_DB_NAME, dbNameStr)
+        elif e.orig[0] == MySqlErr.ER_TABLE_EXISTS_ERROR:
+            if mayExist:
+                return
+            raise DbException(DbException.TB_EXISTS, dbNameStr)
+        else:
+            raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def dropTable(conn, tableName, dbName=None, mustExist=True):
+    """
+    Drop table <tableName> in database <dbName>.
+
+    @param tableName  Table name.
+    @param dbName     Database name.
+    @param mustExist  Flag indicating what to do if the database does not exist.
+
+    If <dbName> is not set, the current database name will be used.
+
+    Raises DbException if the table does not exist and the mustExist flag is set to
+    True, or if MySQL error occurs.
+    """
+    dbNameStr = "`%s`." % dbName if dbName is not None else ""
+    try:
+        conn.execute("DROP TABLE %s`%s`" % (dbNameStr, tableName))
+    except DBAPIError as e:
+        if e.orig[0] == MySqlErr.ER_DB_DROP_EXISTS:
+            raise DbException(DbException.DB_DOES_NOT_EXIST, dbName)
+        elif e.orig[0] == MySqlErr.ER_BAD_TABLE_ERROR:
+            if mustExist:
+                raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+        else:
+            raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def listTables(conn, dbName):
+    """
+    Return list of tables in a given database.
+
+    Raises DbException if MySQL error occurs.
+    """
+    cmd = "SELECT TABLE_NAME FROM information_schema.TABLES "
+    cmd += "WHERE TABLE_SCHEMA='%s'" % dbName
+    try:
+        rows = conn.execute(cmd)
+        return [x[0] for x in rows]
+    except DBAPIError as e:
+        raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def isView(conn, tableName, dbName=None):
+    """
+    Return True if the table <tableName> is a view, False otherwise.
+
+    @param tableName  Table name.
+    @param dbName     Database name.
+
+    @return boolean   True if the table is a view. False otherwise.
+
+    If <dbName> is not set, the current database name will be used.
+
+    Raises DbException if MySQL error occurs.
+    """
+    dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
+    try:
+        rows = conn.execute("SELECT table_type FROM information_schema.tables "
+              "WHERE table_schema=%s AND table_name='%s'" % (dbNameStr, tableName))
+        return rows.first()[0] == 'VIEW'
```

Comment:
This may generate `TypeError` exception if `rows` represent empty result, it may be better to handle it in some way.